core/src/transport/memory.rs

```diff
@@ -184,15 +181,12 @@ impl Transport for MemoryTransport {
         id: ListenerId,
         addr: Multiaddr,
     ) -> Result<(), TransportError<Self::Error>> {
-        let port = if let Ok(port) = parse_memory_addr(&addr) {
-            port
-        } else {
+        let Ok(port) = parse_memory_addr(&addr) else {
             return Err(TransportError::MultiaddrNotSupported(addr));
         };
 
-        let (rx, port) = match HUB.register_port(port) {
-            Some((rx, port)) => (rx, port),
-            None => return Err(TransportError::Other(MemoryTransportError::Unreachable)),
+        let Some((rx, port)) = HUB.register_port(port) else { 
```

Comment:
This could just be `unwrap_or()?`.

Reply:
actually `ok_or` seems to be the better choice

```rust
let (rx, port) = HUB.register_port(port).ok_or(TransportError::Other(MemoryTransportError::Unreachable))?;
```
looks much better I think